src/test/ui/typeck/issue-52082.rs

```diff
@@ -0,0 +1,54 @@
+// Fix issue 52082: Confusing error if accidentially defining a type paramter with the same name as
```

Comment:
Would be helpful to give the filename a bit more context, `issue-52082-type-param-shadows-existing-type.rs`